app/src/main/java/org/mate/state/executables/ActionsScreenState.java

```diff
@@ -410,7 +412,8 @@ public boolean differentColor(IScreenState visitedState) {
                         //MATE.log("compare: " + wThis.getId()+"-"+wThis.getClazz()+": "+wThis.getColor()+"  vs  " + wOther.getId()+"-"+wOther.getClazz()+": " + wOther.getColor());
                         if (!wOther.getColor().equals(wThis.getColor())){
```

Comment:
Use logical operator && to combine the nested if conditions.

Reply:
done